code/src/Consumable.ts

```diff
@@ -0,0 +1,35 @@
+import { Item } from './Item';
+
+export class Consumable extends Item {
+  consumed: boolean = false;
+  spoiled: boolean;
+
+  constructor(name: string, value: number, weight: number, spoiled: boolean) {
+    super(name, value, weight);
+    this.spoiled = spoiled;
+  }
+
+  eat(): string {
+    return 'eat';
+  }
+
+  use(): string {
+    return 'use';
```

Comment:
Incorrect implementation, if a Consumable is not spoiled and is not consumed, calling this should return the value from a call to Consumable.eat().

Reply:
Task is a little confusing, I understand, so maybe my explanation will help: this method is the main method for implementing consuming behavior.
Method "use" should call this.eat() to get the string of "eating". You should check if item is consumed - then return string "you have nothing...". If not consumed, check if it is spoiled - and return either only one string from "eat" or this string + new string with text "you feel sick".